tests/test_lab_dev/test_states.py

```diff
@@ -382,6 +392,16 @@ def test_bargmann_triple_error(self):
         with pytest.raises(AttributeError):
             fock.bargmann
 
+    def test_normalize(self):
+        # Bargmann
+        coh = Coherent([0], x=1.0, y=1.0).dm() * 0.5
+        normalized = coh.normalize()
+        assert np.isclose(normalized.probability, 1.0)
+        # Fock
+        coh.to_fock(5)  # truncated
+        normalized = coh.normalize()
+        assert np.isclose(normalized.probability, 1.0)
+
```

Comment:
If you want to check batched too
```suggestion
    @pytest.mark.parametrize(
        "state",
        [
            0.5 * Coherent([0], x=1.0, y=1.0).dm(),
            0.3 * (Coherent([0], x=1.0, y=1.0).dm() + Coherent([0], x=-1.0, y=-1.0).dm()),
        ],
    )
    def test_normalize(self, state):
        # Bargmann
        normalized = state.normalize()
        assert np.isclose(normalized.probability, 1.0)
        # Fock
        state.to_fock(5)  # truncated
        normalized = state.normalize()
        assert np.isclose(normalized.probability, 1.0)
```